src/ds3.c

```diff
@@ -1285,6 +1291,22 @@ static bool element_equal(const xmlNodePtr xml_node, const char* element_name){
     return xmlStrcmp(xml_node->name, (const xmlChar*) element_name) == 0;
 }
 
+static uint16_t xml_get_uint16(xmlDocPtr doc, xmlNodePtr child_node) {
+    xmlChar* text;
+    uint16_t size;
+    text = xmlNodeListGetString(doc, child_node->xmlChildrenNode, 1);
+    if (text == NULL) {
+        return 0;
+    }
+    size = atoi((char*)text);
```

Comment:
I think there is a glib version of this specifically for a `uint16_t`.  You may go look and see.